modules/timepoint_list/php/timepoint_list.class.inc

```diff
@@ -84,14 +85,8 @@ class Timepoint_List extends \NDB_Menu
         ServerRequestInterface $request,
         RequestHandlerInterface $handler
     ) : ResponseInterface {
-        $attribute = $request->getAttribute("CandID");
-        if ($attribute !== null) {
-            // FIXME: Update the code to add the singleton here, not the identifier.
-            $this->candID = $attribute;
-        } else {
-            $gets         = $request->getQueryParams();
-            $this->candID = $gets['candID'];
-        }
+        // FIXME: Update the code to add the singleton here, not the identifier.
+        $this->candID = new CandID($request->getAttribute('CandID'));
```

Comment:
@johnsaigle this is my last point of concern, if you are removing the if/else, whats handling the logic now ?

Reply:
I'm not sure... I don't know where the old lowercase `candID` (vs `CandID`) was being sent and was relying on the integration tests to catch it. But there wasn't any error. I'm wondering if the if-else was a hack to manage a typo from somewhere else which has since been fixed.

Reply:
IIRC it was to handle both `$URL/$candid/` and `$URL/main.php?test_name=timepoint_list&candID=$candid`, but it was a while ago so I might be misremembering..

Reply:
Ok thanks. Are these paths still a concern?

Reply:
If there is I think we should fix the URL to point to the `$URL/$candid` one.. if that's working (and I'm not misremembering why it was needed), I don't think it should be a concern.

Reply:
given that its going to major which means it will undergo a full round of testing... im fine with it